status/k8s.go

```diff
@@ -311,97 +321,171 @@ retry:
+
+				if err != nil {
+					if _, ok := status.PodState[defaults.HubbleUIDeploymentName]; !ok {
+						status.AddAggregatedWarning(defaults.HubbleUIDeploymentName, defaults.HubbleUIDeploymentName, fmt.Errorf("hubble ui is not deployed"))
+					} else {
+						status.AddAggregatedError(defaults.HubbleUIDeploymentName, defaults.HubbleUIDeploymentName, err)
+						status.CollectionError(err)
+					}
+				}
 
-	disabled, err = k.deploymentStatus(ctx, status, defaults.RelayDeploymentName)
-	status.SetDisabled(defaults.RelayDeploymentName, defaults.RelayDeploymentName, disabled)
-	if err != nil {
-		if _, ok := status.PodState[defaults.RelayDeploymentName]; !ok {
-			status.AddAggregatedWarning(defaults.RelayDeploymentName, defaults.RelayDeploymentName, fmt.Errorf("hubble relay is not deployed"))
-		} else {
-			status.AddAggregatedError(defaults.RelayDeploymentName, defaults.RelayDeploymentName, err)
-			status.CollectionError(err)
-		}
+				// pod status for UI is only validated if the deployment exists
+				if _, ok := status.PodState[defaults.HubbleUIDeploymentName]; ok {
+					err = k.podStatus(ctx, status, defaults.HubbleUIDeploymentName, "k8s-app=hubble-ui", nil)
+					if err != nil {
+						status.CollectionError(err)
+					}
+				}
+
+				return nil
+			},
+		},
+		statusTask{
+			name: defaults.ClusterMeshDeploymentName,
+			task: func(_ context.Context) error {
+				disabled, err := k.deploymentStatus(ctx, status, defaults.ClusterMeshDeploymentName)
+				status.SetDisabled(defaults.ClusterMeshDeploymentName, defaults.ClusterMeshDeploymentName, disabled)
+				if err != nil {
+					if _, ok := status.PodState[defaults.ClusterMeshDeploymentName]; !ok {
+						status.AddAggregatedWarning(defaults.ClusterMeshDeploymentName, defaults.ClusterMeshDeploymentName, fmt.Errorf("clustermesh is not deployed"))
+					} else {
+						status.AddAggregatedError(defaults.ClusterMeshDeploymentName, defaults.ClusterMeshDeploymentName, err)
+						status.CollectionError(err)
+					}
+				}
+
+				// pod status for relay is only validated if the deployment exists
+				if _, ok := status.PodState[defaults.ClusterMeshDeploymentName]; ok {
+					err = k.podStatus(ctx, status, defaults.ClusterMeshDeploymentName, "k8s-app=clustermesh-apiserver", nil)
+					if err != nil {
+						status.CollectionError(err)
+					}
+				}
+
+				return nil
+			},
+		},
+		statusTask{
+			name: defaults.HubbleUIDeploymentName,
+			task: func(_ context.Context) error {
+				err := k.podCount(ctx, status)
+				if err != nil {
+					status.CollectionError(err)
+				}
+
+				return nil
+			},
+		},
 	}
 
-	// pod status for relay is only validated if the deployment exists
-	if _, ok := status.PodState[defaults.RelayDeploymentName]; ok {
-		err = k.podStatus(ctx, status, defaults.RelayDeploymentName, "k8s-app=hubble-relay", nil)
-		if err != nil {
-			status.CollectionError(err)
+	var mutex sync.Mutex
```

Comment:
What is this mutex protecting? From its use it seems it's to avoid concurrent access to `(*Status).CiliumStatus`. As I read it, this will synchronize across all pod status task. Is this intended? Might make sense to add a `sync.Mutex` field to the `Status` type instead?

Reply:
It is intended yes. Do you mean an additional one besides this one https://github.com/cilium/cilium-cli/pull/706/files#diff-2b43f8f754e7e61ebc349444f31f588e20353ee7e69b174c7cfbaad101a6d0ebR104 or the same?

That would also be fine yes.